spec/models/orchestration_template_spec.rb

```diff
@@ -36,6 +36,24 @@
         expect(@existing_record).not_to eq(OrchestrationTemplate.find_or_create_by_contents(@query_hash)[0])
         expect(OrchestrationTemplate.count).to eq(2)
       end
+
+      it "uses subclass if type is present" do
+        expect(ManageIQ::Providers::Vmware::CloudManager::OrchestrationTemplate).to receive(:calc_md5).at_least(:once)
```

Comment:
why `at_lease(:once)`? can it be called more than once?

Reply:
In fact it gets called twice:

- 1st time in L46 `md5s << klass.calc_md5(with_universal_newline(hash[:content]))`
- 2nd time in L56 `template = create!(hash.except(:md5))` because of this hook https://github.com/manageiq/manageiq/blob/master/app/models/orchestration_template.rb#L64

Theoretically I could test for `:twice` but would prefer to keep it more general, should you agree.

Reply:
Oh, yes, it is called twice. Then I think `at_least(:once)` is ok.